src/main/java/com/loofah/graph/api/services/SkillService.java

```diff
@@ -46,7 +49,15 @@ public SkillDTO getById(String id) {
     private SkillDTO transformToSkillDTO(Skill skill) {
         Category category = retrieveCategoryForSkill(skill);
         Grade grade = retrieveGradeForSkill(skill);
-        return new SkillDTO(skill, category, grade);
+        List<Craft> crafts = retrieveCraftsForSkill(skill);
+        return new SkillDTO(skill, category, grade, crafts);
+    }
+
+    private List<Craft> retrieveCraftsForSkill(Skill skill) {
+        if (skill.getCraftTitles().isEmpty()) {
+            throw new GraphQLException("craftTitle is empty for skill with id ["+skill.getTitle()+"]");
```

Comment:
Could the error message say "for skill with title" instead of "id" ?

Reply:
good spot - i've gone for `"craftTitle is empty for skill with id ["+skill.getId()+"]"` because title aka sub-category won't be unique